### PR TITLE
Add `keyLoc` and `valueLoc` to `ValueDefinition`

### LINKED ISSUE #80: Add nodes/locations to value definition parts
Ideally we have a node/location for each value in the value definition, while also having the node/location for the property-key and property-value.

Currently we get the range for the whole value of everything after `static values = `

### PR DESCRIPTION
**Pull request by nachiket87**
fix #80 

Adds keyLoc and valueLoc to the Value Definitions

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export type ValueDefinitionValue = string | number | bigint | boolean | object |
 export type ValueDefinition = {
   type: string
   default: ValueDefinitionValue
+  loc?: any
```
Comment (marcoroth) [inline src/types.ts:14]: Let's use a proper type for this, something like:
```suggestion
  loc?: Acorn.SourceLocation | null,
```

File: src/util/ast.ts [hunk 2]
```diff
@@ -73,13 +74,18 @@ export function convertObjectExpressionToValueDefinition(objectExpression: Acorn
       break
   }
 
-  if (!type) return
+  if (!type) {
+    return
+  } else {
+    loc = typeProperty?.value.loc
+  }
```
Comment (marcoroth) [inline src/util/ast.ts:81]: I think you can remove the `let` on L65 and just use this here:

```suggestion
  if (!type) return

 let loc = typeProperty?.value.loc
```

File: src/util/ast.ts [hunk 3]
```diff
@@ -73,13 +74,18 @@ export function convertObjectExpressionToValueDefinition(objectExpression: Acorn
...
 
   let defaultValue = getDefaultValueFromNode(defaultProperty?.value)
 
   return {
     type,
     default: defaultValue,
+    loc: loc
```
Comment (marcoroth) [inline src/util/ast.ts:88]: This is a great start, thanks for working on this!

The `loc` you have right now represented the value part of the key/value pair. Ideally we have the location information for both the key and the value so we can target both individually.

For example, if you have the same Stimulus Value name twice, we could only target the "key" part, but if you have a wrong value definition we'd want to target the value location.

Maybe we can add something like:

```js
{
  type,
  default: defaultValue,
  keyLoc,
  valueLoc,
}
```

Does that make sense?

**Review by marcoroth — COMMENTED** [review]
This is great @nachiket87, thanks for giving this a go! I have a few comments, but this looks great already!

File: src/util/ast.ts (hunk 3 above)
Comment (nachiket87) [reply src/util/ast.ts:88]: Yes that makes sense I'll work on it asap!

Comment (marcoroth) [reply src/util/ast.ts:88]: Awesome, thank you! Please let me know if you get stuck or have questions! 🙏🏼

**Review by marcoroth — APPROVED** [review]
Awesome, this is perfect! Thank you @nachiket87!

**Comment by marcoroth** [comment]
@nachiket87 just published `v0.1.4` including this change and bumped the version in `stimulus-lsp`.

**Pull request merged**